// ng-amp/bundles/ng-amp.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core')) :
    typeof define === 'function' && define.amd ? define('ng-amp', ['exports', '@angular/core'], factory) :
    (global = global || self, factory(global['ng-amp'] = {}, global.ng.core));
}(this, (function (exports, i0) { 'use strict';

    var NgAmpService = /** @class */ (function () {
        function NgAmpService() {
        }
        return NgAmpService;
    }());
    NgAmpService.ɵprov = i0.ɵɵdefineInjectable({ factory: function NgAmpService_Factory() { return new NgAmpService(); }, token: NgAmpService, providedIn: "root" });
    NgAmpService.decorators = [
        { type: i0.Injectable, args: [{
                    providedIn: 'root'
                },] }
    ];
    NgAmpService.ctorParameters = function () { return []; };

    var NgAmpComponent = /** @class */ (function () {
        function NgAmpComponent() {
        }
        NgAmpComponent.prototype.ngOnInit = function () {
            var _this = this;
            if (document.querySelector('#ng-amp')) {
                return;
            }
            var scriptTag = document.createElement('script');
            scriptTag.id = 'ng-amp';
            scriptTag.src = '//amp.azure.net/libs/amp/2.3.5/azuremediaplayer.min.js';
            scriptTag.onload = function () { return _this.onLoadInit(); };
            var linkTag = document.createElement('link');
            linkTag.rel = 'stylesheet';
            linkTag.href = '//amp.azure.net/libs/amp/2.3.5/skins/amp-default/azuremediaplayer.min.css';
            document.body.appendChild(scriptTag);
            document.head.insertBefore(linkTag, document.head.firstChild);
        };
        NgAmpComponent.prototype.onLoadInit = function () {
            var config = {
                controls: this.controls !== false,
                autoplay: this.autoplay,
                poster: this.poster,
                width: this.width,
                height: this.height,
                fluid: this.fluid,
                logo: { enabled: false },
            };
            var component = amp(this.videoPlayer.nativeElement, config);
            component.src([{
                    src: this.src,
                    type: 'application/vnd.ms-sstr+xml'
                }]);
            this.videoPlayer.ngComponent = component;
        };
        return NgAmpComponent;
    }());
    NgAmpComponent.decorators = [
        { type: i0.Component, args: [{
                    selector: 'ng-amp',
                    template: "\n    <video #video class=\"azuremediaplayer amp-default-skin\" controls>\n      <p class=\"amp-no-js\">\n          To view this video please enable JavaScript, and consider upgrading to a web browser that supports HTML5 video.\n      </p>\n    </video>\n  "
                },] }
    ];
    NgAmpComponent.ctorParameters = function () { return []; };
    NgAmpComponent.propDecorators = {
        videoPlayer: [{ type: i0.ViewChild, args: ['video',] }],
        src: [{ type: i0.Input }],
        controls: [{ type: i0.Input }],
        autoplay: [{ type: i0.Input }],
        poster: [{ type: i0.Input }],
        width: [{ type: i0.Input }],
        height: [{ type: i0.Input }],
        fluid: [{ type: i0.Input }]
    };

    var NgAmpModule = /** @class */ (function () {
        function NgAmpModule() {
        }
        return NgAmpModule;
    }());
    NgAmpModule.decorators = [
        { type: i0.NgModule, args: [{
                    declarations: [NgAmpComponent],
                    imports: [],
                    exports: [NgAmpComponent]
                },] }
    ];

    /*
     * Public API Surface of ng-amp
     */

    /**
     * Generated bundle index. Do not edit.
     */

    exports.NgAmpComponent = NgAmpComponent;
    exports.NgAmpModule = NgAmpModule;
    exports.NgAmpService = NgAmpService;

    Object.defineProperty(exports, '__esModule', { value: true });

})));
