// ng-amp/fesm2015/ng-amp.js
import { ɵɵdefineInjectable, Injectable, Component, ViewChild, Input, NgModule } from '@angular/core';

class NgAmpService {
    constructor() { }
}
NgAmpService.ɵprov = ɵɵdefineInjectable({ factory: function NgAmpService_Factory() { return new NgAmpService(); }, token: NgAmpService, providedIn: "root" });
NgAmpService.decorators = [
    { type: Injectable, args: [{
                providedIn: 'root'
            },] }
];
NgAmpService.ctorParameters = () => [];

class NgAmpComponent {
    constructor() { }
    ngOnInit() {
        if (document.querySelector('#ng-amp')) {
            return;
        }
        const scriptTag = document.createElement('script');
        scriptTag.id = 'ng-amp';
        scriptTag.src = '//amp.azure.net/libs/amp/2.3.5/azuremediaplayer.min.js';
        scriptTag.onload = () => this.onLoadInit();
        const linkTag = document.createElement('link');
        linkTag.rel = 'stylesheet';
        linkTag.href = '//amp.azure.net/libs/amp/2.3.5/skins/amp-default/azuremediaplayer.min.css';
        document.body.appendChild(scriptTag);
        document.head.insertBefore(linkTag, document.head.firstChild);
    }
    onLoadInit() {
        const config = {
            controls: this.controls !== false,
            autoplay: this.autoplay,
            poster: this.poster,
            width: this.width,
            height: this.height,
            fluid: this.fluid,
            logo: { enabled: false },
        };
        const component = amp(this.videoPlayer.nativeElement, config);
        component.src([{
                src: this.src,
                type: 'application/vnd.ms-sstr+xml'
            }]);
        this.videoPlayer.ngComponent = component;
    }
}
NgAmpComponent.decorators = [
    { type: Component, args: [{
                selector: 'ng-amp',
                template: `
    <video #video class="azuremediaplayer amp-default-skin" controls>
      <p class="amp-no-js">
          To view this video please enable JavaScript, and consider upgrading to a web browser that supports HTML5 video.
      </p>
    </video>
  `
            },] }
];
NgAmpComponent.ctorParameters = () => [];
NgAmpComponent.propDecorators = {
    videoPlayer: [{ type: ViewChild, args: ['video',] }],
    src: [{ type: Input }],
    controls: [{ type: Input }],
    autoplay: [{ type: Input }],
    poster: [{ type: Input }],
    width: [{ type: Input }],
    height: [{ type: Input }],
    fluid: [{ type: Input }]
};

class NgAmpModule {
}
NgAmpModule.decorators = [
    { type: NgModule, args: [{
                declarations: [NgAmpComponent],
                imports: [],
                exports: [NgAmpComponent]
            },] }
];

/*
 * Public API Surface of ng-amp
 */

/**
 * Generated bundle index. Do not edit.
 */

export { NgAmpComponent, NgAmpModule, NgAmpService };
